test/cmd/debug.sh

```diff
@@ -280,3 +280,131 @@ run_kubectl_debug_baseline_node_tests() {
   set +o nounset
   set +o errexit
 }
+
+run_kubectl_debug_restricted_tests() {
+  set -o nounset
+  set -o errexit
+
+  create_and_use_new_namespace  
+  kube::log::status "Testing kubectl debug profile restricted"
+
+  ### Pod Troubleshooting by ephemeral containers with restricted profile
+  
+  # Pre-Condition: Pod "nginx" is created
+  kubectl run target "--image=${IMAGE_NGINX:?}" "${kube_flags[@]:?}"
+  kube::test::get_object_assert pod "{{range.items}}{{${id_field:?}}}:{{end}}" 'target:'
+  # Restricted profile just works in not restricted namespace
+  # Command: add a new debug container with restricted profile
+  output_message=$(kubectl debug target -it --image=busybox --attach=false -c debug-container --profile=restricted "${kube_flags[@]:?}")
+  kube::test::if_has_not_string "${output_message}" 'forbidden: violates PodSecurity'
+  # Post-Conditions
+  kube::test::get_object_assert pod/target '{{range.spec.ephemeralContainers}}{{.name}}:{{end}}' 'debug-container:'
+  # Clean up
+  kubectl delete pod target "${kube_flags[@]:?}"
+
+  local ns_name="namespace-restricted"
+  # Command: create namespace and add a label
+  kubectl create namespace "${ns_name}"
+  kubectl label namespace "${ns_name}" pod-security.kubernetes.io/enforce=restricted
+  output_message=$(kubectl get namespaces "${ns_name}" --show-labels)
+  kube::test::if_has_string "${output_message}" 'pod-security.kubernetes.io/enforce=restricted'
+  kubectl config set-context "${CONTEXT}" --namespace="${ns_name}"
```

Comment:
Would be better to not change the default context. Instead you can pass `-n ns_name` to the commands below.